qCC/ccColorFromScalarDlg.cpp

```diff
@@ -43,13 +43,27 @@ ccColorFromScalarDlg::ccColorFromScalarDlg(QWidget* parent, ccPointCloud* pointC
 	, m_ui(new Ui::ColorFromScalarDialog)
 {
 	m_ui->setupUi(this);
-	
+	this->setAttribute(Qt::WA_DeleteOnClose, true);
+	if (m_cloud->getCurrentDisplayedScalarFieldIndex() == -1)
```

Comment:
I'm still expecting a sanity check on the fact that the cloud has at least one scalar field (I'm not comfortable with making too much assumptions on what has been done by the caller)

Reply:
I'm open to suggestion on what to do in the case of a failure, the test is easy enough, but on any failure the only way to handle it will be for the caller not to show the dialog at all after construction, which relies on assuming the caller does the error check on the dialog.

Reply:
Maybe put an error at the top of the dialog explaining the problem and disable everything except the Close button?

Reply:
Oh it's very defensive, and only to prevent a crash. I would simply return right away (the dialog won't work, that's perfect).

Reply:
So an assert() for debug mode, and an error to the console so there's some feedback in a release build? (like below)

Reply:
I went through and added error checks and on unrecoverable errors the dialog has all elements other than the close button, I don't display the warning on the dialog itself but ccLog::Error creates a popup and a console message.

I found another bug while going through this, fixed wouldn't work on initial dialog display when the SF Min and Max prevented the default fixed value.

Reply:
I'd say another look at this is required because I was poking around all over the place and changed some order of operations to support the error handling, and I spent a bunch of time trying to track down that fixed value bug, so yeah I'd like another look from you guys :)

Reply:
I guess it's ok (even if you've pushed the logic quite far ;). In the case there's no SF, I would have simply not connected the signals / slots so that nothing happens and you don't have to test everywhere that the system is valid... But at least now you have "belt and suspenders" (don't know if that translates ;)

Reply:
That for sure translates, I get accused of that pretty regularly 😀